File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int st = -1,mid = 0, end = nums.size()-1;
         while(mid<=end){
            if(nums[mid] == 0){
                st++;
               swap(nums[st],nums[mid]);
               mid++;
            }
            else if(nums[mid] == 2){
                swap(nums[mid],nums[end]);
                end--;
            }
            else{
                mid++;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {

    void swap(int a , int b, int[] nums){
        int temp = nums[a];
        nums[a] = nums[b];
        nums[b] = temp;
    }
    public void sortColors(int[] nums) {
        int st = -1; int mid = 0; int end = nums.length-1;

        while(mid <= end){
           if(nums[mid] == 0){
            st++;
            swap(st, mid, nums);
            mid++;
            
            
           }
           else if(nums[mid] == 2){
            swap(mid, end, nums);
            end--;
           }
           else mid++;
        }
    }
}